webbpsf/tests/test_opds.py

```diff
@@ -22,3 +29,49 @@ def test_enable_adjustable_ote():
 
     assert ote.rms() > rms*10, "Huge piston offset didn't make the WFE much worse"
 
+
+
+GLOBAL_FOCUS = [-0.018251043541410904e-08]
+GLOBAL_FOCUS_SCALED = [-0.018251043541410904e-08 * 0.5]
+COEFFS_A1 = np.array([-3.52633363e-09, -2.90050902e-09, 1.25432196e-09, -7.43319098e-12,
+                      -5.82462948e-11, -1.27115922e-10, -1.91541104e-12, 3.64760396e-11,
+                      4.97176630e-13])
+thermal_model_parameters = ([1 * u.day, 'SM', GLOBAL_FOCUS, None],
+                            [1 * u.day, 'SM', GLOBAL_FOCUS_SCALED, .5],
+                            [1440, 'SM', GLOBAL_FOCUS, None],
+                            [1 * u.day, 'A1', COEFFS_A1, None],
+                            [0.0 * u.day, 'SM', [0.0], None],
+                            [0.0 * u.day, 'A1', np.zeros(9), None],
+                            [1.0 * u.day, 'D1', [0.0], None])
+@pytest.mark.parametrize('time, seg, coeff_truth, scaling', thermal_model_parameters)
+def test_thermal_slew_opd(time, seg, coeff_truth, scaling):
+    """ Test that the OTE Thermal model is outputting the correct values """
+    delta_time = time
+    # Create the thermal model
+    otelm = webbpsf.opds.OTE_Linear_Model_WSS()
+    otelm.delta_time = delta_time
+    otelm.start_angle = -5.
+    otelm.end_angle = 45.
+    otelm.scaling = scaling
+    otelm.thermal_model = webbpsf.opds.OteThermalModel(otelm.delta_time)
+    coeffs = otelm.thermal_slew_opd(segid=seg)
+    # Pull out coefficients
+    if isinstance(coeffs, float):
+        coeffs = [coeffs]
+    # Assert the coefficents
+        for coeff, truth in zip(coeffs, coeff_truth):
+            assert np.round(coeff, decimals=4) == np.round(truth, decimals=4)
+
+
+def test_thermal_slew_update_opd():
+    ''' Test that running webbpsf.opds.OTE_Linear_Model_WSS.thermal_slew() will
+        give the expected output'''
+    otelm = webbpsf.opds.OTE_Linear_Model_WSS()
+    otelm.thermal_slew(delta_time=1.0*u.day)
+    assert np.max(otelm.opd) == 4.1333852193439e-08
```

Comment:
This test is a dangerous one, since it is looking for exact floating point equality. Finite floating point precision (~ "rounding errors") can give different behavior on different machines, in terms of the last few decimal places. It's better to write tests like this in a way that checks for closeness (within some tolerance) rather than exact equality in every bit. The `np.isclose` function is good for this. It's also good form to give test assertions an informative error message in case the assertion fails.  
```
assert np.isclose(np.max(otelm.opd), 4.13338e-08), "OPD max does not match expected value after 1 day slew."
```

Reply:
Thanks! I will make this change.